src/index.ts

```diff
@@ -81,43 +88,133 @@ function arrayEquals<T>(as: T[], bs: T[]): boolean {
 
-    if (!lookup.has(lookupModel)) {
-        lookup.set(lookupModel, []);
+    if (!externalConvertersLookup.has(lookupModel)) {
+        externalConvertersLookup.set(lookupModel, []);
     }
 
-    // key created above as needed
-    if (!lookup.get(lookupModel)!.includes(definition)) {
-        lookup.get(lookupModel)!.splice(0, 0, definition);
+    // key created above
+    if (!externalConvertersLookup.get(lookupModel)!.includes(definition)) {
+        externalConvertersLookup.get(lookupModel)!.splice(0, 0, definition);
     }
 }
 
-function removeFromLookup(zigbeeModel: string | undefined, definition: Definition): void {
+function removeFromExternalConvertersLookup(zigbeeModel: string | undefined, definition: IndexedDefinition): void {
     const lookupModel = zigbeeModel ? zigbeeModel.toLowerCase() : 'null';
 
-    if (lookup.has(lookupModel)) {
-        const i = lookup.get(lookupModel)!.indexOf(definition);
+    if (externalConvertersLookup.has(lookupModel)) {
+        const i = externalConvertersLookup.get(lookupModel)!.indexOf(definition);
 
         if (i > -1) {
-            lookup.get(lookupModel)!.splice(i, 1);
+            externalConvertersLookup.get(lookupModel)!.splice(i, 1);
+        }
+
+        if (externalConvertersLookup.get(lookupModel)!.length === 0) {
+            externalConvertersLookup.delete(lookupModel);
+        }
+    }
+}
+
+export function getFromExternalConvertersLookup(zigbeeModel: string | undefined): IndexedDefinition[] | undefined {
+    const lookupModel = zigbeeModel ? zigbeeModel.toLowerCase() : 'null';
+
+    if (externalConvertersLookup.has(lookupModel)) {
+        return externalConvertersLookup.get(lookupModel);
+    }
+
+    return externalConvertersLookup.get(lookupModel.replace(/\0(.|\n)*$/g, '').trim());
+}
+
+export function removeExternalDefinitions(converterName?: string): void {
+    for (let i = 0; i < externalDefinitionsCount; i++) {
+        const definition = externalDefinitions[i];
+
+        if (converterName && definition.externalConverterName !== converterName) {
+            continue;
+        }
+
+        if ('zigbeeModel' in definition && definition.zigbeeModel) {
+            for (const zigbeeModel of definition.zigbeeModel) {
+                removeFromExternalConvertersLookup(zigbeeModel, definition);
+            }
+        }
+
+        if ('fingerprint' in definition && definition.fingerprint) {
+            for (const fingerprint of definition.fingerprint) {
+                removeFromExternalConvertersLookup(fingerprint.modelID, definition);
+            }
+        }
+
+        externalDefinitions.splice(i, 1);
+
+        externalDefinitionsCount--;
+        i--;
+    }
+}
+
+export function addDefinition(definition: IndexedDefinition): void {
+    externalDefinitions.splice(0, 0, definition);
+
+    if (definition.externalConverterName) {
+        externalDefinitionsCount++;
+    }
+
+    if ('fingerprint' in definition && definition.fingerprint) {
```

Comment:
```suggestion
    if (definition.fingerprint) {
```
Should be fine too, definition doesn't have a fingerprint property it will eval to false.

Reply:
I missed those, too much copy/pasting... 😝

Reply:
Fixed the ones that aren't used for narrowing 😉